src/components/Campaign.tsx

```diff
@@ -17,18 +16,21 @@ export const editorTopBarTestId = "editor-top-bar-message";
 
 export const Campaign = () => {
   const { idCampaign } = useParams() as Readonly<{ idCampaign: string }>;
-  const { setContent, unsetContent, getContent } = useSingletonEditor();
-
   const campaignContentQuery = useGetCampaignContent(idCampaign);
   const campaignContentMutation = useUpdateCampaignContent();
+  const { getContent, save } = useSingletonEditor(
+    {
+      initialContent: campaignContentQuery.data,
+      onSave: async () => {
+        const content = await getContent();
+        campaignContentMutation.mutate({ idCampaign, content });
+      },
+    },
+    [campaignContentQuery.data]
+  );
```

Comment:
Not sure how often Campaing re renders. 
Maybe we can prevent extra calls to useSingletonEditor if we control initialContent, onSave, or deps are not changing? have you considered to use useCallback or useMemo on this?

Reply:
No estoy seguro de haber entendido. Donde harías el cambio?

Reply:
`campaignContentQuery.data` just change one time when _campaign_ is rendered, on the other hand useSingleton contain an effect.
https://github.com/FromDoppler/doppler-editors-webapp/blob/376cda39f0f42346030d7724712ebcb51af927d4/src/components/SingletonEditor.tsx#L56

That's why I didn't think it necessary to use a memo or callback.

Reply:
Entiendo, creo que está bien tenerlo en cuenta, cómo dice @Cromeror, de ser necesario, estaría bien hacer algo esto dentro del custom hook.

Reply:
Tenia mis dudas al respecto asi que realicé unas pruebas y no encontré mejora significativa en memorizar lo que le pasamos a useSingletonEditor porque se realiza la misma cantidad de renderizados sobre _Campaign_ y considero que la lógica de _onSave_ no es compleja para requerir memorizar.

La prueba consistió en contabilizar la cantidad de renderizados usando _useCallback_ y _useMemo_, el resultado es igual a cuando no se usan.

Reply:
Genial, si no hay mejora perceptible no se aconseja implementar esa estrategia.